Source/WebCore/PAL/pal/text/TextEncodingRegistry.cpp

```diff
@@ -254,8 +255,19 @@ std::unique_ptr<TextCodec> newTextCodec(const TextEncoding& encoding)
     Locker locker { encodingRegistryLock };
 
     ASSERT(textCodecMap);
+    if (!encoding.isValid()) {
+        RELEASE_LOG_ERROR(TextEncoding, "Trying to create new text codec with invalid (null) encoding name. Will default to UTF-8.");
+        return TextCodecUTF8::codec();
+    }
     auto result = textCodecMap->find(encoding.name());
-    ASSERT(result != textCodecMap->end());
+    if (result == textCodecMap->end()) {
+        RELEASE_LOG_ERROR(TextEncoding, "Can't find codec for valid encoding %s . Will default to UTF-8.", encoding.name());
```

Comment:
You'll want to use `... for valid encoding %" PUBLIC_LOG_STRING ". will default...` to make sure the string shows in sysdiagnoses.